File: app/src/main/java/com/wang/example/msg/bean/ImgBean.java
package com.wang.example.msg.bean;

import androidx.annotation.NonNull;

import com.alibaba.fastjson.JSON;
import com.wang.container.adapter.IContainerItemAdapter;
import com.wang.example.msg.adapter.ImgAdapter;

public class ImgBean extends BaseMsgBean {
    public ImgDataEntity imgInfo;

    protected ImgBean(BaseMsgBean oldBean) {
        super(oldBean);
        imgInfo = JSON.parseObject(oldBean.data, ImgDataEntity.class);
    }

    public static class ImgDataEntity {
        public int imgRes;
    }

    @NonNull
    @Override
    public Class<? extends IContainerItemAdapter> getBindAdapterClass() {
        return ImgAdapter.class;
    }
}

File: app/src/main/java/com/wang/example/msg/bean/ImgBean.kt
package com.wang.example.msg.bean

import com.google.gson.Gson
import com.wang.example.msg.adapter.ImgAdapter

class ImgBean(oldBean: BaseMsgBean) : BaseMsgBean(oldBean) {
    var imgInfo: ImgDataEntity

    class ImgDataEntity {
        var imgRes = 0
    }

    override fun getBindAdapterClass() = ImgAdapter::class.java

    init {
        imgInfo = Gson().fromJson(oldBean.data, ImgDataEntity::class.java)
    }
}